Accessibility Enhancement

MuriloDiasTCD opened the pull request:
### Accessibility & UI/UX Enhancement

The focus of this PR is improve the accessibility of user with no access privileges (not signed users) targeting level AA of [WCAG 2.1 ](https://www.w3.org/TR/WCAG21/) with **_exclusions*_**, and improve page elements for a more pleasant navigation Without modify the location of the elements. 

The Automated tool used to check HTML tags and best practices is the well known Chrome plugin [Lighthouse](https://chromewebstore.google.com/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk), and for manual tests the [screen reader](https://chromewebstore.google.com/detail/screen-reader/kgejglhpjiefppelpmljglcjbhoiplfn) and [WCAG 2.1 ](https://www.w3.org/TR/WCAG21/).

It is important to note that users with privilege will have the experience improved as well as there are shared elements between both users as Tables, JS code and gadgets. 

The picture 1 helps to illustrate the accessibility improvement: 
![image](https://github.com/Digital-Repository-of-Ireland/dri-app/assets/115575268/24a35cc3-4151-489d-a473-a40aa9406989)
**_Picture 1: Accessibility Improvement by Lighthouse tool._**

Code changes proposed in this PR are mostly related to:
- Color contrast
- Font Size
- Element Size
- Hover interactions
- HTML tags
- ARIA properties
- Shape of HTML elements
- Fix of HTML elements hierarchy

Follow a preview of the altered webpages:
![image](https://github.com/Digital-Repository-of-Ireland/dri-app/assets/115575268/18cd2a71-1fe8-4ab0-9dcf-ebe0f7db1dce)
**_Picture 2: DRI REPOSITORY > Browse by Data & Collections > Collections_**

![image](https://github.com/Digital-Repository-of-Ireland/dri-app/assets/115575268/886b83ab-65ef-4ab0-bd06-7e52a9a10b09)
_**Picture 3: DRI REPOSITORY > Browse by Data & Collections > Collections > Collection**_

![image](https://github.com/Digital-Repository-of-Ireland/dri-app/assets/115575268/8e5bc959-64aa-40e1-868b-d06204979122)
_**Picture 4: DRI REPOSITORY > Browse by Data & Collections > Objects**_

![image](https://github.com/Digital-Repository-of-Ireland/dri-app/assets/115575268/628132e7-2ea3-45ed-bfba-59536b07efe3)
_**Picture 5: DRI REPOSITORY > Browse by Data & Collections > Objects > Object**_

![image](https://github.com/Digital-Repository-of-Ireland/dri-app/assets/115575268/26248759-651a-4d07-beff-a56b8c3935e3)
_**Picture 6: DRI REPOSITORY > Browse by Organization**_

**_Exclusions_**
The accessibility exclusions are related to material that user upload. An asset can be uploaded by partner and in case of missing significant information or in non-standard text format is found at the metadata (misspelling, symbols, languages), as ARIA relies in metadata to created tags for screen readers.
Another source of non-accessible information are assets uploaded by user as **images**, **PDF**, **video**, **interactive resource**, **software**, **dataset** and **3D** with immersive experience.

stkenny commented:
Social media button alignment doesn't seem correct
![Screenshot from 2024-01-24 17-37-37](https://github.com/Digital-Repository-of-Ireland/dri-app/assets/1716730/97e55453-4260-4eb7-83a7-d77c94776bef)

Is black on red an accessible combination?
![Screenshot from 2024-01-24 17-37-58](https://github.com/Digital-Repository-of-Ireland/dri-app/assets/1716730/309e327f-2a7d-48ac-8e98-c8b5a8b233e8)

Advanced search modal buttons incorrect
![Screenshot from 2024-01-24 17-38-21](https://github.com/Digital-Repository-of-Ireland/dri-app/assets/1716730/f8635237-622f-48ff-8333-ca142b5b0653)

Modal fixes from develop seem to have been lost, e.g. buttons not correct, two close icons. This was fixed on develop but lost on merge
![Screenshot from 2024-01-24 17-40-15](https://github.com/Digital-Repository-of-Ireland/dri-app/assets/1716730/2f264c61-b4ae-4978-9052-05945098cd66)

MuriloDiasTCD commented:
> Social media button alignment doesn't seem correct ![Screenshot from 2024-01-24 17-37-37](https://private-user-images.githubusercontent.com/1716730/299407755-97e55453-4260-4eb7-83a7-d77c94776bef.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYxOTM3MzEsIm5iZiI6MTcwNjE5MzQzMSwicGF0aCI6Ii8xNzE2NzMwLzI5OTQwNzc1NS05N2U1NTQ1My00MjYwLTRlYjctODNhNy1kNzdjOTQ3NzZiZWYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDEyNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMjVUMTQzNzExWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MjdlYzc0NTM2MzU2MzhjNDg1NzE2ZTY5NjE3NTQ5NmMwZTU4NWMwNWRlMjFkN2MzZmEyMzMzOWM2MzlhNTFkYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.mRwW-yZuEomQykbvihE70oaN0UUJ2jTbcQoK_4tV6qw)
> 
**Answer: Alignment CSS included.**


> Is black on red an accessible combination? ![Screenshot from 2024-01-24 17-37-58](https://private-user-images.githubusercontent.com/1716730/299408198-309e327f-2a7d-48ac-8e98-c8b5a8b233e8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYxOTM3MzEsIm5iZiI6MTcwNjE5MzQzMSwicGF0aCI6Ii8xNzE2NzMwLzI5OTQwODE5OC0zMDllMzI3Zi0yYTdkLTQ4YWMtOGU5OC1jOGI1YThiMjMzZTgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDEyNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMjVUMTQzNzExWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ODI4NTdkYjQ3ZjRiNzE3Nzc0ZWRkNjExMTk4NDA3OTBiMjk5ZmJjMjJmMGRjNDhhMTE3NGM2YmU1YTM5NDVjZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.GQO1b2oV0yaB4Atx18LD5HSy0aeus5bPL6-HOWA2O9A)
> 
**Answer: Yes, the selected red color has more than 4:5 tones from black or white.**
![image](https://github.com/Digital-Repository-of-Ireland/dri-app/assets/115575268/03c6fbd0-9a03-49a3-b29d-38f970626439)
![image](https://github.com/Digital-Repository-of-Ireland/dri-app/assets/115575268/73c0dc89-4eb5-4893-88c6-ff9a1400a27e)


> Advanced search modal buttons incorrect ![Screenshot from 2024-01-24 17-38-21](https://private-user-images.githubusercontent.com/1716730/299408332-f8635237-622f-48ff-8333-ca142b5b0653.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYxOTM3MzEsIm5iZiI6MTcwNjE5MzQzMSwicGF0aCI6Ii8xNzE2NzMwLzI5OTQwODMzMi1mODYzNTIzNy02MjJmLTQ4ZmYtODMzMy1jYTE0MmI1YjA2NTMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDEyNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMjVUMTQzNzExWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OGI2NTZkYjgyNjFmN2QyMmEwMTEwNGRjODZmNWU4MDU0NDc2OWU5MGI2OTJkNDE4NjdmN2UzY2Y0NjNiYzk1NyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.KeTyf81jnEWieanoJ5oRcwB9x7tVwgz5yGT3GgTTWV8)
> 
**Answer: Modified todri_button_black:**
![image](https://github.com/Digital-Repository-of-Ireland/dri-app/assets/115575268/1f9f305c-3ad1-418c-bb71-8ced24325da1)


> Modal fixes from develop seem to have been lost, e.g. buttons not correct, two close icons. This was fixed on develop but lost on merge ![Screenshot from 2024-01-24 17-40-15](https://private-user-images.githubusercontent.com/1716730/299408645-2f264c61-b4ae-4978-9052-05945098cd66.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYxOTM3MzEsIm5iZiI6MTcwNjE5MzQzMSwicGF0aCI6Ii8xNzE2NzMwLzI5OTQwODY0NS0yZjI2NGM2MS1iNGFlLTQ5NzgtOTA1Mi0wNTk0NTA5OGNkNjYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDEyNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMjVUMTQzNzExWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MjMwNGIxYWQ3ZGU2MDZjNWYyOTY2MjNiZTA4ZWUyNmY4ZGJjMGYzMzBkOTM4NGViYzI3YzEwMjc0NDI5MmUwNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.LtCNQ1nmGiggjwHh3XmC0Sgfsd6EtKaSIsrsNF4KxuI)
**Answer: Updated button. And branch updated from my end (Metadata working fine) If still have problem from your end please let me know.**

NOTE: 
I've tested on lighthouse and it's 100% accessible.
But when testing with WAVE, it present 3 errors related to accessibility. It repeats on all pages, do you know if it's a necessary piece of code or if we can get rid of it? 
The source of the error are the 3 empty buttons, but code seems to be useless (I can be wrong).

```
...
<!-- close dri_wrapper-->
<div id="cboxOverlay" style="display: none"></div>
<div id="colorbox" class="" role="dialog" tabindex="-1" style="display: none">
    <div id="cboxWrapper">
        <div>
            <div id="cboxTopLeft" style="float: left"></div>
            <div id="cboxTopCenter" style="float: left"></div>
            <div id="cboxTopRight" style="float: left"></div>
        </div>
        <div style="clear: left">
            <div id="cboxMiddleLeft" style="float: left"></div>
            <div id="cboxContent" style="float: left">
                <div id="cboxTitle" style="float: left"></div>
                <div id="cboxCurrent" style="float: left"></div>
                <button type="button" id="cboxPrevious"></button
                ><button type="button" id="cboxNext"></button
                ><button type="button" id="cboxSlideshow"></button>
                <div id="cboxLoadingOverlay" style="float: left"></div>
                <div id="cboxLoadingGraphic" style="float: left"></div>
            </div>
            <div id="cboxMiddleRight" style="float: left"></div>
        </div>
        <div style="clear: left">
            <div id="cboxBottomLeft" style="float: left"></div>
            <div id="cboxBottomCenter" style="float: left"></div>
            <div id="cboxBottomRight" style="float: left"></div>
        </div>
    </div>
    <div
        style="
            position: absolute;
            width: 9999px;
            visibility: hidden;
            display: none;
            max-width: none;
        "
    ></div>
</div>
...
```

MuriloDiasTCD commented:
> The colorbox code is needed. It only needs to be on the object show page though. We might want to look at alternatives as the library no longer seems to be updated. The specific issue is [jackmoore/colorbox#765](https://github.com/jackmoore/colorbox/issues/765) I can look to address this in a separate issue. This PR already has a lot of changes.

Understood, thanks for explaining Stuart.
The source of the accessibility issue is because the 3 buttons has no text in it. I've tried to put some char or string in it for the sake of accessibility, but seems like it's auto generated. 

`<button type="button" id="cboxPrevious"> </button>
<button type="button" id="cboxNext"></button>
<button type="button" id="cboxSlideshow"></button>`
